the-collab-lab/tcl-15-smart-shopping-list
# #issue-9: Filter the shopping Lists

## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    |  🔨  Refactoring  |

## Description
**This PR adds new functionality to the app and refactors some of the previous code**

**First**: It adds a text filed and a clear-text-field button in `List.js` to allow users to enter or then clear search terms.
**Second**: It declares the `getItemsKeys(shoppingList, searchTerm)` function in `List.js` which takes a `shoppingList` and `searchTerm` and returns the items' keys according to that data. 
**Third**: It creates the `SortedList` component for displaying the items on the user's shopping list and refactors a few other things.

## Related Issue
Closes #9 

## Acceptance Criteria

## Update
Users can filter their shopping lists by entering search terms into the text field added in the `List` route, and the list will only display the items whose names include these search terms. *If nothing matches, nothing shows*.

## How things look so far
![a gif of how users use the app to search for items](https://i.ibb.co/CBtKKqY/search.gif)


## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin hn-dg-issue-9` 
* Check that branch out with `git checkout hn-dg-issue-9`
* Run `yarn start` to run the [local server](http://localhost:3000/).
* Join a list by using the **bushy ret morale** token, or create your own list and add some items to it.
* If you're not already, go to the `List` route and use the text field to search items on the list.

Now, you **can** filter the shopping list by anything you enter into the text field. If there are no items whose names include your search term, nothing should show up.

Thank you!
